src/module/item/base/document.ts

```diff
@@ -948,4 +973,9 @@ interface RefreshFromCompendiumParams {
     update?: boolean;
 }
 
+/** An extension of DatabaseUpdateOperation to pass on system specific data between phases */
+type ItemPF2eDatabaseUpdateOperation = DatabaseUpdateOperation<foundry.abstract.Document | null> & {
+    previousSystemData?: { maxHitPoints?: number };
```

Comment:
Maybe just call it `previous` or `prior` and include an outer `system` property.

Reply:
We can omit the system property. I called it that to try to distinguish from core foundry properties, but then after sleeping I kinda realized that we'd just put pf2e in the name if we really want to do that. previous is fine!